saleor/webhook/payloads.py

```diff
@@ -1080,3 +1159,178 @@ def generate_excluded_shipping_methods_for_checkout_payload(
+    requestor: Optional["RequestorOrLazyObject"] = None,
+    with_meta: bool = True,
+):
+    manager = get_plugins_manager()
+    lines = OrderLine.objects.prefetch_related("variant__product__product_type")
+
+    return _generate_order_payload(
+        order,
+        requestor,
+        with_meta,
+        order_prices_data=_generate_order_prices_data_with_taxes(order, manager, lines),
+        order_lines_payload=_generate_order_lines_payload_with_taxes(
+            order, manager, lines
+        ),
+        included_taxes_in_price=include_taxes_in_prices(),
+    )
+
+
+def generate_order_payload_without_taxes(
+    order: "Order",
+    requestor: Optional["RequestorOrLazyObject"] = None,
+    with_meta: bool = True,
+):
+    lines = OrderLine.objects.prefetch_related("variant__product__product_type")
+    included_taxes_in_price = include_taxes_in_prices()
+
+    return _generate_order_payload(
+        order,
+        requestor,
+        with_meta,
+        order_prices_data=_generate_order_prices_data_without_taxes(
+            order, included_taxes_in_price
+        ),
+        order_lines_payload=_generate_order_lines_payload_without_taxes(
+            order, lines, included_taxes_in_price
+        ),
+        included_taxes_in_price=included_taxes_in_price,
+    )
+
+
+def _generate_checkout_prices_data_with_taxes(
+    manager: PluginsManager,
+    checkout_info: CheckoutInfo,
+    lines: Iterable[CheckoutLineInfo],
+) -> Dict[str, Decimal]:
+    subtotal = checkout_calculations.checkout_subtotal(
+        manager=manager,
+        checkout_info=checkout_info,
+        lines=lines,
+        address=None,
+    )
+    total = checkout_calculations.checkout_total(
+        manager=manager,
+        checkout_info=checkout_info,
+        lines=lines,
+        address=None,
+    )
+
+    def qp(amount: Decimal) -> Decimal:
+        return quantize_price(amount, checkout_info.checkout.currency)
+
+    return {
+        "subtotal_net_amount": qp(subtotal.net.amount),
+        "subtotal_gross_amount": qp(subtotal.gross.amount),
+        "total_net_amount": qp(total.net.amount),
+        "total_gross_amount": qp(total.gross.amount),
+    }
+
+
+def _generate_checkout_prices_data_without_taxes(
+    checkout: Checkout,
+    included_taxes_in_price: bool,
+) -> Dict[str, Decimal]:
+    def untaxed_price_amount(price: TaxedMoney) -> Decimal:
+        return quantize_price(
+            get_base_price(price, included_taxes_in_price), checkout.currency
+        )
+
+    return {
+        "subtotal_base_amount": untaxed_price_amount(checkout.subtotal),
+        "total_base_amount": untaxed_price_amount(checkout.total),
+    }
+
+
+def generate_checkout_payload(
+    checkout: "Checkout",
+    requestor: Optional["RequestorOrLazyObject"] = None,
+):
+    manager = get_plugins_manager()
+    lines, _ = fetch_checkout_lines(checkout, prefetch_variant_attributes=True)
+    checkout_info = fetch_checkout_info(checkout, lines, [], manager)
```

Comment:
Shouldn't we pass discounts to this function?

Reply:
Resolved in recent commit